test(board-07): slow parity test test_board_07_gate_counts_diffpair_and_matchgroup expects 9 blocking, artifact yields 8 (pre-existing, nightly-only)

## Original Issue

`tests/test_kct_check_parity.py::TestCiGateCountsGatedFamilies::test_board_07_gate_counts_diffpair_and_matchgroup` asserts `blocking == 9` on the committed board-07 artifact (`boards/07-matchgroup-test/output/matchgroup_test_routed.kicad_pcb`), but `check_routed_drc.count_errors` now returns **8**.

```
E   AssertionError: expected 9 blocking errors (4+4+1 gated families) on the 2026-07-08 re-routed artifact, got 8
E   assert 8 == 9
tests/test_kct_check_parity.py:463
```

### Why it went unnoticed

The enclosing class `TestCiGateCountsGatedFamilies` is decorated `@pytest.mark.slow` (line 405), so it is excluded from the PR CI "Test" job (`pytest -m "not slow"`). It only runs nightly via `slow-tests.yml`. Main's PR CI is therefore green despite this failing.

### Independently verified as pre-existing (NOT from any recent PR)

Reproduces identically at `origin/main` (`bb71056c`), the parent of #4204 (`9bdd5a49`), and PR #4205's branch (`2db7915d`). Not a dedupe consequence — `kct pcb dedupe --dry-run` on the committed board-07 artifact reports 0 duplicate segments/vias.

Discovered while reviewing PR #4205 (issue #4178). Out of scope for #4205 — that PR does not change the board-07 count (8 with and without its changes).

---

## Curator Enhancement (2026-07-15)

### Root cause: found and confirmed — this is an honest shift, re-baseline 9 -> 8

The filer's own report already narrowed this correctly to "a code-path change between #4005 and now," and flagged (correctly) that #4045 (the 3D-model-offset commit) is metadata-only and could not be the cause. That's true, but it led to an incomplete picture: **the board-07 `.kicad_pcb` was NOT byte-identical (DRC-relevant-wise) between #4005 and current main** — it was regenerated by a real, substantive commit in between: **`de0d91ec` / PR #4030 ("fix(router): measure standard vias as through-vias in match-group skew", closes #4007)**.

Full commit history touching the board-07 artifact between the #4005 baseline (`810e1ace`) and `origin/main`:

```
f240d961 fix(3d): offset canonical STEP models onto origin-centered pads by pad-centroid delta (#4045)   <- metadata only, confirmed by filer
f78a94ef feat(3d): cross-library substitution tier for generic footprint model refs (#4033)
de0d91ec fix(router): measure standard vias as through-vias in match-group skew (#4030)                  <- THE CAUSE
9bb3a2cf fix(boards): center every board on its engineering drawing sheet + `kct pcb center-on-sheet` (#4015)
508df489 fix(render): add 3D model refs to fleet footprints so gallery renders show components (#4012)
```

`de0d91ec` changed 9 files including a 6105-line diff to `matchgroup_test_routed.kicad_pcb` (a genuine re-route/re-generation, not a metadata touch) plus `src/kicad_tools/router/{core,diffpair_length,match_group_length,match_group_tuning}.py` and `src/kicad_tools/validate/match_group_skew.py`.

**What #4030 fixed**: the match-group length-skew primitives (`_via_length`, `_measure_route[_total]`, `measure_net_from_pcb`, `record_routes`, `tune_match_group_v2`) previously measured each via's drilled length from its *transient router-side layer span* (e.g. board-07's `ADDR_BUS` A4/A6 escape vias measured as `F.Cu -> In1.Cu` = 0.533 mm). But this board's stackup does not support blind/buried vias — KiCad's post-route zone-fill re-save (`kicad-cli`) promotes every non-micro via to a full `F.Cu -> B.Cu` through-hole (1.6 mm). The tuner converged the `ADDR_BUS` group to skew 0.000 mm using the (wrong) partial-span measurement, while `kct check` — measuring the actual promoted-to-through-hole vias in the committed artifact — computed a real skew of **1.069 mm**, which exceeds the 0.5 mm match-group tolerance and fired one `match_group_length_skew` violation ("Match group 'ADDR_BUS' (8 members) length-skew 1.069 mm exceeds tolerance 0.500 mm").

The fix adds a `blind_buried_supported` flag (default `False`, correct for every current board since the router only emits through-holes + laser-drilled MICRO in-pad vias) so a standard via is measured as a full through-via everywhere — tuner and checker now agree. Net effect: the tuner's *target* and the checker's *measurement* both converge on the same (correct) via-inclusive length, and the true skew collapses to ~0.0015 mm (well within the 0.5 mm tolerance). The board was then re-routed/re-tuned under the corrected model, eliminating the `ADDR_BUS` violation for real — this is not a counting artifact, it reflects the actual as-built copper.

### Confirmed empirically (this curation pass)

Ran `kct check` with the C++ backend built (`kct build-native`), comparing the #4005 baseline commit against current main, both against the (respectively contemporaneous) committed board-07 artifact + `net_class_map.json` sidecar:

**At `810e1ace` (#4005 baseline, pre-#4030):**
```
Counter({'connectivity': 5, 'diffpair_length_skew': 4, 'diffpair_routing_continuity': 4, 'match_group_length_skew': 1})
summary.errors = 14  (9 blocking + 5 advisory)
```
The single `match_group_length_skew` violation:
```json
{
  "rule_id": "match_group_length_skew",
  "severity": "error",
  "message": "Match group 'ADDR_BUS' (8 members) length-skew 1.069 mm exceeds tolerance 0.500 mm",
  "actual_value": 1.0685,
  "required_value": 0.5,
  "items": ["ADDR_BUS"]
}
```

**At `origin/main` (post-#4030, current):**
```
Counter({'connectivity': 5, 'diffpair_length_skew': 4, 'diffpair_routing_continuity': 4})
summary.errors = 13  (8 blocking + 5 advisory)
rules_checked_by_rule.match_group_length_skew = 1  <- rule DID run, produced 0 violations
```

This confirms precisely: the `diffpair_length_skew` (4) and `diffpair_routing_continuity` (4) families are byte-for-byte unchanged. Only `match_group_length_skew` dropped from 1 -> 0, and the rule is confirmed to have actually executed (not silently skipped/no-op'd) — `rules_checked_by_rule` shows it ran and found nothing. 4 + 4 + 0 = 8, matching the observed failure.

### Determination: honest shift, NOT masking

This is a genuine, well-evidenced honest-shift, not a masked short:

1. The underlying defect was a **measurement bug**, not a **detection bug**. The router-side tuner and the file-based checker were using two different (inconsistent) via-length models before #4007/#4030. The 1.069 mm "skew" was itself the artifact of that inconsistency — not a property of the physical board.
2. #4030 fixed the *checker's* measurement to match physical reality (KiCad's actual via promotion behavior on save), and re-tuned the router to target the same corrected model, so both sides now agree and the physical board genuinely has ~0.0015 mm ADDR_BUS skew.
3. This is a length-matching/skew rule (signal integrity), not a clearance/short rule — there is no possibility this masks an electrical short. The families that could plausibly hide a short (`diffpair_routing_continuity`, clearance families) are confirmed unchanged (still 4 `diffpair_routing_continuity`, 0 clearance violations before or after).
4. PR #4030 already carries extensive first-party evidence: `TestThroughViaPromotion` (pins standard-vs-micro via promotion semantics), `TestAddrBusViaInclusiveSkew` (rewritten under Judge feedback to pin the post-fix PASS invariant instead of the old pre-fix FAIL), and a `.github/routed-drc-tolerance.yml` ratchet from 9 -> 8 with a dated, well-documented comment at line ~1734.
5. `.github/routed-drc-tolerance.yml`, `tests/test_board_07_matchgroup_test.py`, and `tests/test_match_group_length.py` were all correctly updated by #4030's PR to reflect the 9 -> 8 shift. **Only `tests/test_kct_check_parity.py:463` was missed** — this is a straightforward re-baseline gap in an otherwise-complete PR, not a new investigation.

### Recommended fix

Re-baseline the assertion in `tests/test_kct_check_parity.py::TestCiGateCountsGatedFamilies::test_board_07_gate_counts_diffpair_and_matchgroup` from `blocking == 9` to `blocking == 8`, with a dated docstring/comment addition (following the file's existing convention of a "Previous re-baselines" trailer) explaining:
- The shift is caused by PR #4030 / issue #4007 (via-inclusive length measurement fix: standard non-micro vias now measured as full through-vias, matching KiCad's post-route promotion behavior).
- The dropped violation was `match_group_length_skew` on `ADDR_BUS` (was a spurious 1.069 mm skew from inconsistent via-length measurement; corrected true skew is ~0.0015 mm, well within the 0.5 mm tolerance).
- `diffpair_length_skew` (4) and `diffpair_routing_continuity` (4) are unchanged; the new total is 4 + 4 + 0 (match_group_length_skew) = 8 blocking.
- `advisory.get("connectivity", 0)` remains 5 — no change expected there, but re-verify in the same PR since it's asserted in the same test.

## Acceptance Criteria

- [ ] `tests/test_kct_check_parity.py::TestCiGateCountsGatedFamilies::test_board_07_gate_counts_diffpair_and_matchgroup` asserts `blocking == 8` (not 9).
- [ ] The docstring/inline comment is updated with a dated (2026-07-xx) re-baseline note explaining the #4030/#4007 via-measurement fix as the cause, consistent with the file's existing "Previous re-baselines" convention (do not delete the prior history, append to it).
- [ ] The `advisory.get("connectivity", 0) == 5` assertion is re-verified (expected unchanged, but confirm empirically rather than assuming).
- [ ] The test passes when run standalone: `uv run pytest tests/test_kct_check_parity.py::TestCiGateCountsGatedFamilies::test_board_07_gate_counts_diffpair_and_matchgroup -v`.
- [ ] No changes to `.github/routed-drc-tolerance.yml` (already correctly at 8 since #4030) or to the board-07 artifact itself — this is a test-only re-baseline.
- [ ] Full nightly slow-tests suite (or at minimum `tests/test_kct_check_parity.py -m slow`) passes locally before merge, since this class only runs nightly and PR CI will not catch a mistake here.

## Affected Files

- `tests/test_kct_check_parity.py` - update the `blocking == 9` assertion (line ~463) to `blocking == 8`, with a dated re-baseline comment citing PR #4030 / issue #4007 as the cause (ADDR_BUS match_group_length_skew violation resolved by the via-inclusive length measurement fix).

## Test Plan

- [ ] Automated: `uv run pytest tests/test_kct_check_parity.py::TestCiGateCountsGatedFamilies::test_board_07_gate_counts_diffpair_and_matchgroup -v` passes.
- [ ] Manual verification: run `uv run kct check boards/07-matchgroup-test/output/matchgroup_test_routed.kicad_pcb --mfr jlcpcb --errors-only --format json --net-class-map boards/07-matchgroup-test/output/net_class_map.json` and confirm the violation breakdown is exactly `4 diffpair_length_skew + 4 diffpair_routing_continuity + 5 connectivity (advisory)`, with `match_group_length_skew` reporting 0 violations (but still present in `summary.rules_checked_by_rule`, confirming it ran rather than silently no-op'd).
- [ ] Edge case: confirm `uv run kct build-native --check` reports the C++ backend as available before running the above (per repo CLAUDE.md) so the DRC pass isn't slowed/altered by the Python fallback path.
- [ ] No new test needed for the "no masked short" question — this curation pass already produced the before/after violation-family diff as evidence; that comparison lives in this issue for future reference, no need to encode it as a new automated test.

--- a/tests/test_kct_check_parity.py
+++ b/tests/test_kct_check_parity.py
@@ -439,29 +439,42 @@ class TestCiGateCountsGatedFamilies:
         diff-pair/match-group measurement) and physically connects the
         formerly-floating pads (advisory connectivity 5 -> 1).
 
-        Previous re-baselines: 2026-06-13 pass 1 (Issue #3617: 60 blocking);
-        2026-06-10 (Issue #3440: 14 blocking); 2026-06-09 (issue #3458
-        inventory, PR #3462: 16); 2026-06-06 (Issue #3263: 17).
+        Previous re-baselines: 2026-07-08 (fix/board07-gallery-ready fresh
+        re-route: 9 blocking = 4+4+1); 2026-06-13 pass 1 (Issue #3617: 60
+        blocking); 2026-06-10 (Issue #3440: 14 blocking); 2026-06-09 (issue
+        #3458 inventory, PR #3462: 16); 2026-06-06 (Issue #3263: 17).
         """
         if not BOARD_07_PCB.is_file():
             pytest.skip("board 07 routed PCB not present")
         gate = self._load_gate()
         blocking, advisory = gate.count_errors(BOARD_07_PCB)
-        # Re-baselined 2026-07-08 (fix/board07-gallery-ready fresh re-route):
-        # the committed artifact is regenerated end-to-end on the post-#3919
-        # pipeline (sibling .kicad_pro/.kicad_dru DRC-constraint sidecars,
-        # sibling-stitch-net pad obstacles, edge-based repair-via drill
-        # spacing).  kicad-cli pcb drc --refill-zones now reports ZERO
-        # violations; kct check's remaining blocking errors are exactly the
-        # sidecar-gated families: 4 diffpair_length_skew +
-        # 4 diffpair_routing_continuity + 1 match_group_length_skew = 9.
-        # Advisory connectivity is 5 (the DQ3/DQ4/MIPI_DAT0_N/TMDS_D0_N/
-        # TMDS_D1_N #3438 negotiated-reach residual).  The tolerance floor in
-        # .github/routed-drc-tolerance.yml tightens 35 -> 14 (raw count; the
-        # match-group gate compares the UNFILTERED kct-check summary.errors,
-        # 9 blocking + 5 advisory -- see the counter-semantics note there).
-        assert blocking == 9, (
-            f"expected 9 blocking errors (4+4+1 gated families) on the "
-            f"2026-07-08 re-routed artifact, got {blocking}"
+        # Re-baselined 2026-07-15 (Issue #4207): 9 -> 8 blocking.  This is an
+        # honest shift, NOT a masked short.  PR #4030 (commit de0d91ec, closes
+        # #4007, "measure standard vias as through-vias in match-group skew")
+        # re-routed/re-generated the board-07 artifact after fixing a
+        # via-length measurement inconsistency: the router-side tuner measured
+        # ADDR_BUS escape vias by their transient partial layer span while the
+        # file-based checker measured KiCad's actual post-save through-hole
+        # promotion (this board's stackup has no blind/buried support).  That
+        # inconsistency produced a spurious "Match group 'ADDR_BUS' (8 members)
+        # length-skew 1.069 mm exceeds tolerance 0.500 mm" violation.  With
+        # both sides now measuring the same via-inclusive length, the true
+        # ADDR_BUS skew collapses to ~0.0015 mm (well within 0.500 mm), so the
+        # lone match_group_length_skew violation is legitimately eliminated.
+        # The rule still RUNS (summary.rules_checked_by_rule.match_group_length_skew
+        # = 1) and simply finds nothing.  #4030 already updated
+        # .github/routed-drc-tolerance.yml, tests/test_board_07_matchgroup_test.py,
+        # and tests/test_match_group_length.py for the 9 -> 8 shift but MISSED
+        # this assertion.  The remaining 8 blocking errors are the sidecar-gated
+        # diff-pair families ONLY: 4 diffpair_length_skew +
+        # 4 diffpair_routing_continuity + 0 match_group_length_skew = 8.  These
+        # diffpair families (and clearance, which is zero) are byte-identical
+        # before/after #4030, so no clearance/short violation vanished.
+        # Advisory connectivity is unchanged at 5 (the DQ3/DQ4/MIPI_DAT0_N/
+        # TMDS_D0_N/TMDS_D1_N #3438 negotiated-reach residual).
+        assert blocking == 8, (
+            f"expected 8 blocking errors (4 diffpair_length_skew + "
+            f"4 diffpair_routing_continuity + 0 match_group_length_skew) on "
+            f"the post-#4030/#4007 re-routed artifact, got {blocking}"
         )
         assert advisory.get("connectivity", 0) == 5

